main.go

```diff
@@ -88,6 +90,18 @@ func startExtension() (*Worker, bool) {
 		}
 		log.Printf("Found lambda tags: %v", tags)
 		config.Tags = tags
+
+		functionConfig, err := awsClient.GetFunctionConfiguration(functionARN)
+		if err != nil {
+			log.Printf("Failed to get Lambda Function Configuration, err: %v", err)
+			lambdaClient.InitError(ctx, extensionID, lambda.ClientError, lambda.LambdaError{
+				Type:    "GetFunctionConfigurationError",
+				Message: err.Error(),
+			})
+			return nil, false
+		}
+		config.Tags[lambda.ProcessRuntimeNameTag] = *functionConfig.Runtime
+		config.Tags[lambda.RuntimeArchitectureTag] = utils.GetRuntimeArchitecture()
```

Comment:
don't we have this in getfunctionconfiguration response?
https://docs.aws.amazon.com/lambda/latest/dg/API_GetFunctionConfiguration.html

Reply:
getfunctionconfiguration response has an array for `architectures`. I was not sure but I thought that it returns supported architectures for the function. That's why I did not use it.

Reply:
![image](https://github.com/edgedelta/edgedelta-lambda-extension/assets/111270176/8ade035b-f41b-4fe8-8dc7-45324b5b8a21)
It does not return architecture in golang. I'm not sure why but GetFunctionConfiguration does not event have `Architectures` field

Reply:
https://github.com/aws/aws-sdk-go/blob/main/service/lambda/api.go#L12069

it has architectures field. you can check the sdk version, and update if necessary.

Reply:
instead of using listtags and getfunctionconfiguration, we can use getfunction which has both.

https://github.com/aws/aws-sdk-go/blob/main/service/lambda/api.go#L13387